l3kernel/l3seq.dtx

```diff
@@ -189,6 +189,45 @@
 %   \end{texnote}
 % \end{function}
 %
+% \begin{function}[added = 2024-11-26]
+%   {
+%     \seq_set_regex_extract:Nnn, \seq_set_regex_extract:cnn,
+%     \seq_set_regex_extract:NNn, \seq_set_regex_extract:cNn,
+%     \seq_gset_regex_extract:Nnn, \seq_gset_regex_extract:cnn,
+%     \seq_gset_regex_extract:NNn, \seq_gset_regex_extract:cNn,
+%   }
+%   \begin{syntax}
+%     \cs{seq_set_regex_extract:Nnn} \meta{seq~var} \Arg{regular expression} \Arg{token list}
+%     \cs{seq_set_regex_extract:NNn} \meta{seq~var} \Arg{compiled regex} \Arg{token list}
+%   \end{syntax}
+%   Finds the first match of the \meta{regular expression} in the
+%   \meta{token list}. If it exists, the match is stored as the first
+%   item of the \meta{seq~var}, and further items are the contents of
+%   capturing groups, in the order of their opening parenthesis. If
+%   there is no match, the \meta{seq~var} is cleared.
+% \end{function}
+%
+% \begin{function}[added = 2024-11-26]
+%   {
+%     \seq_set_regex_split:Nnn, \seq_set_regex_split:cnn,
+%     \seq_set_regex_split:NNn, \seq_set_regex_split:cNn,
+%     \seq_gset_regex_split:Nnn, \seq_gset_regex_split:cnn,
+%     \seq_gset_regex_split:NNn, \seq_gset_regex_split:cNn,
+%   }
+%   \begin{syntax}
+%     \cs{seq_set_regex_split:Nnn} \meta{seq~var} \Arg{regular expression} \Arg{token list}
+%     \cs{seq_set_regex_split:NNn} \meta{seq~var} \Arg{compiled regex} \Arg{token list}
```

Comment:
(general comment)

About naming consistency: l3regex uses `{<regex>}` and `<regex var>` respectively. Especially `\Arg{compiled regex}` cannot denote an `N`-type arg.

And, do we need a reference to the l3regex module chapter at the end? Like "See documentation of regular expressions in Chapter~\ref{...} for more info."

Reply:
that might be a good idea in the 3 places

Reply:
> `l3regex` isn't entirely consistent - it uses `<regular expression` in all of the descriptions even where the code example uses `<regex>`

(Unrelated to this PR.)

I'd thought about documenting full and short forms of expressions and variables in some place(s), for example in the list of all data types, in sec. 1.1 "Naming functions and variables", and perhaps also at the beginning of chapters of corresponding modules.